### PR TITLE
Enhancement #7202 done

### PR DESCRIPTION
**Pull request by rachana-webkul**
## Issue Reference
#7202
>Option added for upload category banner

### CODE REVIEW

**Comment by Sanjaybhattwebkul** [comment]
@Rachana1005 
**Not working,** please find the screenshot [Here](https://prnt.sc/8_orbmh7WAgo)

**Comment by Rachana1005** [comment]
@Sanjaybhattwebkul 
Please check the changes. It's done from my side

**Comment by Sanjaybhattwebkul** [comment]
@Rachana1005 
Now this is **working fine**, please find the screenshot [Here](https://prnt.sc/VwCAwJmmKcm4)

**Review by Sanjaybhattwebkul — APPROVED** [review]

**Comment by Sanjaybhattwebkul** [comment]
@Rachana1005 
**Improve the UI** of the category banner in the **admin panel**, please take a reference from [Here](https://prnt.sc/prCqderpyxJx)

File: packages/Webkul/Admin/src/Resources/views/catalog/categories/create.blade.php [hunk 1]
```diff
@@ -102,6 +102,18 @@
...
+
+                                <image-wrapper button-label="{{ __('admin::app.catalog.products.add-image-btn-title') }}" input-name="category_banner" :multiple="false"></image-wrapper>
+
+                                <span class="control-error" v-if="{!! $errors->has('image.*') !!}">
+                                    @foreach ($errors->get('image.*') as $key => $message)
+                                        @php echo str_replace($key, 'Image', $message[0]); @endphp
+                                    @endforeach
+                                </span>
+
+                                <span class="control-info mt-10">{{ __('admin::app.catalog.categories.image-size') }}</span>
```
Comment (vivek-webkul) [inline packages/Webkul/Admin/src/Resources/views/catalog/categories/create.blade.php:116]: @Rachana1005 
Need to mention the correct suggestion size for the banner image. Here you used the suggestion of the Image size.

File: packages/Webkul/Admin/src/Resources/views/catalog/categories/edit.blade.php [hunk 2]
```diff
@@ -140,6 +140,17 @@
...
+                                <label>{{ __('admin::app.catalog.categories.category_banner') }}</label>
+                                <image-wrapper button-label="{{ __('admin::app.catalog.products.add-image-btn-title') }}" input-name="category_banner" :multiple="false" :images='"{{ $category->banner_url }}"'></image-wrapper>
+
+                                <span class="control-error" v-if="{!! $errors->has('image.*') !!}">
+                                    @foreach ($errors->get('image.*') as $key => $message)
+                                        @php echo str_replace($key, 'Image', $message[0]); @endphp
+                                    @endforeach
+                                </span>
+
+                                <span class="control-info mt-10">{{ __('admin::app.catalog.categories.image-size') }}</span>
```
Comment (vivek-webkul) [inline packages/Webkul/Admin/src/Resources/views/catalog/categories/edit.blade.php:153]: @Rachana1005 
Need to mention the correct suggestion size for the banner image. Here you used the suggestion of the Image size.

File: packages/Webkul/Category/src/Database/Migrations/2023_01_16_163456_add_category_banner_to_categories_table.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('categories', function (Blueprint $table) {
+            $table->text('category_banner')->nullable();
```
Comment (vivek-webkul) [inline packages/Webkul/Category/src/Database/Migrations/2023_01_16_163456_add_category_banner_to_categories_table.php:17]: @Rachana1005 
Add column exists check.

File: packages/Webkul/Category/src/Repositories/CategoryRepository.php [hunk 4]
```diff
@@ -73,6 +74,7 @@ public function update(array $data, $id, $attribute = 'id')
         $category->update($data);
 
         $this->uploadImages($data, $category);
+        $this->uploadBanner($data,$category);
```
Comment (vivek-webkul) [inline packages/Webkul/Category/src/Repositories/CategoryRepository.php:77]: @Rachana1005 

Why, we are using new method, we can use the $type param for the banner image.

**Review by vivek-webkul — CHANGES_REQUESTED** [review]

**Comment by Sanjaybhattwebkul** [comment]
@Rachana1005 
The UI part is still not fixed.

**Comment by Sanjaybhattwebkul** [comment]
@Rachana1005 
**Still not fixed** for the admin panel. Please find the **screenshot** [Here](https://prnt.sc/EMJbWbvWGdOg)

**Comment by Rachana1005** [comment]
@Sanjaybhattwebkul 
It's working from my side
Screenshot from 2023-01-18 11-03-57

File: packages/Webkul/Admin/src/Resources/views/catalog/categories/edit.blade.php [hunk 5]
```diff
@@ -140,6 +140,17 @@
                                         @php echo str_replace($key, 'Image', $message[0]); @endphp
                                     @endforeach
                                 </span>
+
+                                <label>{{ __('admin::app.catalog.categories.category_banner') }}</label>
+                                <large-image-wrapper button-label="{{ __('admin::app.catalog.products.add-image-btn-title') }}" input-name="category_banner" :multiple="false" :images='"{{ $category->banner_url }}"'></large-image-wrapper>
+
+                                <span class="control-error" v-if="{!! $errors->has('image.*') !!}">
```
Comment (vivek-webkul) [inline packages/Webkul/Admin/src/Resources/views/catalog/categories/edit.blade.php:147]: Wrong error field check for category_banner.

File: packages/Webkul/Category/src/Repositories/CategoryRepository.php [hunk 6]
```diff
@@ -48,7 +48,8 @@ public function create(array $data)
         $category = $this->model->create($data);
 
         $this->uploadImages($data, $category);
-
+         $this->uploadImages($data, $category, 'category_banner');
```
Comment (vivek-webkul) [inline packages/Webkul/Category/src/Repositories/CategoryRepository.php:51]: No validation(mime type) form post for category_banner image type.

**Review by vivek-webkul — CHANGES_REQUESTED** [review]

**Comment by Sanjaybhattwebkul** [comment]
@Rachana1005 
Now, this i**s working fine,** please find the  screenshot [Here](https://prnt.sc/ZtpPsL_dP7Kz)

**Review by Sanjaybhattwebkul — APPROVED** [review]

**Review by devansh-webkul — CHANGES_REQUESTED** [review]
This pull request is having 155 file changes. Please check.

**Comment by Sanjaybhattwebkul** [comment]
@rachana-webkul 
Any update?

**Comment by Sanjaybhattwebkul** [comment]
@rachana-webkul 
Closing this pull request, because another pull request is created for the same issue.

**Pull request closed without merging**